src/components/network-chart/index.js

```diff
@@ -136,7 +137,9 @@ class NetworkChart extends React.PureComponent<Props> {
             columnCount={1}
             focusable={true}
             specialItems={this._specialItems}
-            containerWidth={3000}
+            containerWidth={width}
+            // We want to force a full rerender when the width and range changes.
+            forceRender={width * 10000 + previewDuration}
```

Comment:
I believe this line is trying to create a unique key based off of the width and previewDuration. Essentially one number needs to be the max reasonable amount value. So either the max reasonable width, or max reasonable previewDuration. `10000px` seems like a reasonable max width, but that means this line should be reversed to.

`width + previewDuration * 10000`.

Also, it would be good to maybe document this a bit better, as it can appear like mysterious magical math.

Reply:
I used a string instead, because I realized that using the duration isn't appropriate; think when we'll have a keyboard shortcut to move the preview forward or backward keeping the same duration. Instead I use a string that concatenates the start and end of the range, and the full width.